File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/ServiceFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class ServiceFileType extends LanguageFileType {
  public static final ServiceFileType INSTANCE = new ServiceFileType();

  private ServiceFileType() {
    super(UnitFileLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Service unit configuration for systemd";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "A unit configuration file whose name ends in \".service\" encodes information about a process controlled and supervised "
           + "by systemd.";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "service";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return UnitFileIcon.FILE;
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/ServiceFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class ServiceFileType private constructor() : AbstractUnitFileType(UnitFileLanguage.INSTANCE) {
  override fun getName(): String {
    return "Service unit configuration for systemd"
  }

  override fun getDescription(): String {
    return displayName
  }

  override fun getDefaultExtension(): String {
    return "service"
  }

  override fun getIcon(): Icon? {
    return UnitFileIcon.SERVICE
  }

  override fun getDisplayName(): @Nls String {
    return "Service unit configuration (systemd)"
  }

  companion object {
    @JvmField
    val INSTANCE = ServiceFileType()
  }
}
